core/rpc/core/src/impl/utils.rs

```diff
@@ -851,12 +851,20 @@ impl<C: CkbRpc> MercuryRpcImpl<C> {
             .occupied_capacity(data_occupied)
             .map_err(|e| CoreError::OccupiedCapacityError(e.to_string()))?;
 
+        let mut occupied = occupied.as_u64();
         // To make CKB `free` represent available balance, pure ckb cell should be spendable.
-        let occupied = if cell.cell_data.is_empty() && cell.cell_output.type_().is_none() {
-            0
-        } else {
-            occupied.as_u64()
-        };
+        if cell.cell_data.is_empty() && cell.cell_output.type_().is_none() {
+            occupied = 0;
+        }
+        // sUDT cell with 0 udt amount should be spendable.
+        if let Some(type_script) = cell.cell_output.type_().to_opt() {
+            let type_code_hash: H256 = type_script.code_hash().unpack();
+            if type_code_hash == **SUDT_CODE_HASH.load() {
+                if self.generate_udt_amount(cell, &io_type).is_zero() {
+                    occupied = 0;
```

Comment:
1. Only occupied of secp sUDT cell with 0 udt amount should be 0, here miss secp judgement.
2. For secp sUDT cell, occupied should be its actual occupied space, not all its capacity. Because the free part of secp udt cell can be used for CKB collection.

Reply:
1. Ok, already added secp lock script check.
2. There is already a judgment on whether udt amout is zero. If it is not zero, occupied will not be modified.